### PR TITLE
Ex 04

### PR DESCRIPTION
**Pull request by pkhetland**
Exercise files for EX 04. Files are approved by PEP-8 standards, and each file contains a main program that runs example instances. All tests in test-file are passed.

Thanks!

### CODE REVIEW

File: src/petter_hetland_ex/ex04/walker.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            self.x -= 1
+
+        self.num_steps += 1
+        return self.x, self.num_steps
+
+    def get_position(self):
+        return self.x
+
+    def is_at_home(self):
+        if self.x == self.h:
```
Comment (ahmedalbuni) [inline src/petter_hetland_ex/ex04/walker.py:28]: you can replace this with a single line: 
return self.x == self.h

File: src/petter_hetland_ex/ex04/myrand.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    r[n+1] = a * r[n] mod m
+
+    where ``a = 7**5 = 16807`` and ``m = 2**31-1``.
+    """
+    def __init__(self, seed):
+        self.init_number = seed
+
+    def rand(self):
+        self.next_number = (16807 * self.init_number) % (2**31-1)
```
Comment (ahmedalbuni) [inline src/petter_hetland_ex/ex04/myrand.py:21]: we use self.variable for variable defined in _init_ method, here you can use next_number as a name

File: src/petter_hetland_ex/ex04/myrand.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+class ListRand:
+    def __init__(self, list):
+        self.list = list
+        self.times_called = 0
+
+    def rand(self):
+        item = self.times_called
+        self.times_called += 1
+        try:
```
Comment (ahmedalbuni) [inline src/petter_hetland_ex/ex04/myrand.py:34]: It's better to check for the given index against the size of the array

**Review by ahmedalbuni — APPROVED** [review]
Good job!

File: src/petter_hetland_ex/ex04/walker.py (hunk 1 above)
Comment (pkhetland) [reply src/petter_hetland_ex/ex04/walker.py:28]: Thanks Ahmed, this is really helpful. Changes made according to comments.

File: src/petter_hetland_ex/ex04/myrand.py (hunk 3 above)
Comment (pkhetland) [reply src/petter_hetland_ex/ex04/myrand.py:34]: Thanks for the advice, Ahmed. Files changed according to comments.

**Pull request merged**